tests/lexer/tminushandling.nim

```diff
@@ -0,0 +1,82 @@
+# Test numeric literals and handling of minus symbol
+
+import std/macros
+import strutils
+import mlexerutils
+
+const one = 1
+const minusOne = `-`(one)
+
+# check when a minus (-) is a negative sign for a literal
+
+doAssert -1 == minusOne:
+  "unable to parse a spaced-prefixed negative int"
+
+doAssert lispReprStr(-1) == """(IntLit -1)""":
+  "failed to include minus sign when lexing integer literal"
+
+doAssert -1.0'f64 == minusOne.float64:
+  "unable to parse a spaced-prefixed negative float"
+
+doAssert lispReprStr(-1.000'f64) == """(Float64Lit -1.0)""":
+  "failed to include minus sign lexing float literal preceded by left paren"
+
+doAssert lispReprStr( -1.000'f64) == """(Float64Lit -1.0)""":
+  "failed to include minus sign lexing float literal preceded by a space"
+
+doAssert [-1].contains(minusOne):
+  "unable to handle negatives after square bracket"
+
+doAssert lispReprStr([-1]) == """(Bracket (IntLit -1))""":
+  "failed to include minus sign lexing int literal preceded by left bracket"
```

Comment:
IMO these messages are just noise; it's clear enough from context; maybe keep the ones you feel are most important for test comprehension and drop the other ones.